torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -749,7 +751,38 @@ Stmt* GPUMetaVarRewriter::mutate(const For* v) {
   return v->cloneWithNewBody(body);
 }
 
+std::vector<Stmt*> GPUMetaVarRewriter::handleDependences(const Block* v) {
+  analysis::MemDependencyChecker analyzer;
+  v->accept(&analyzer);
+
+  // Iterate over the statements in the block. If any statement is dependent on
+  // any previous statement in the block, then insert a SyncThreads before that
+  // statement. Note that the iteration below is quadratic in the number of
+  // statements in the block.
+  auto vStmts = v->stmts();
+  std::vector<Stmt*> stmts(vStmts.begin(), vStmts.end());
+  std::set<Stmt*> stmtsNeedingSync;
+  for (auto it1 = stmts.begin(); it1 != stmts.end(); ++it1) {
+    for (auto it2 = std::next(it1); it2 != stmts.end(); ++it2) {
```

Comment:
You probably want it2 to start at it1 to cover the case where there's a hazard between a stmt and the same stmt from another loop iteration.

Reply:
How are hazards within the same stmt from different iterations meant to be handled? We can't be inserting syncThreads right?

Reply:
Per our discussion offline, I am not changing this.